modules/015-admission-policy-engine/hooks/storage_class_change.go

```diff
@@ -16,13 +16,22 @@ limitations under the License.
 
 package hooks
 
-import "github.com/deckhouse/deckhouse/go_lib/hooks/storage_class_change"
+import (
+	"github.com/flant/addon-operator/pkg/module_manager/go_hook"
+
+	"github.com/deckhouse/deckhouse/go_lib/hooks/storage_class_change"
+)
 
 var _ = storage_class_change.RegisterHook(storage_class_change.Args{
-	ModuleName:         "admission-policy-engine",
-	Namespace:          "d8-admission-policy-engine",
-	LabelSelectorKey:   "app",
-	LabelSelectorValue: "trivy-provider",
-	ObjectKind:         "StatefulSet",
-	ObjectName:         "trivy-provider",
+	ModuleName:                    "admissionPolicyEngine",
+	Namespace:                     "d8-admission-policy-engine",
+	LabelSelectorKey:              "app",
+	LabelSelectorValue:            "trivy-provider",
+	ObjectKind:                    "StatefulSet",
+	ObjectName:                    "trivy-provider",
+	D8ConfigStorageClassParamName: "denyVulnerableImages.storageClass",
+	BeforeHookCheck: func(input *go_hook.HookInput) bool {
+		return input.Values.Get("admissionPolicyEngine.denyVulnerableImages.enabled").Bool() ||
+			input.ConfigValues.Get("admissionPolicyEngine.denyVulnerableImages.enabled").Bool()
+	},
```

Comment:
```suggestion
		return input.Values.Get("admissionPolicyEngine.denyVulnerableImages.enabled").Bool()
	},
```

you don't need to check `ConfigValues` because they are inherited by `Values` as is

Reply:
With the help of [this](https://github.com/deckhouse/deckhouse/blob/main/modules/015-admission-policy-engine/openapi/values.yaml#L1) syntax